idea/idea-fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/WrapWithSafeLetCallFactories.kt

```diff
@@ -0,0 +1,336 @@
+        familyAndActionName(KotlinBundle.lazyMessage("wrap.with.let.call"))
+        applyTo { targetExpression, input ->
+            val nullableExpression = input.nullableExpressionPointer.element ?: return@applyTo
+            if (!nullableExpression.parents.contains(targetExpression)) {
+                LOG.warn(
+                    "Unexpected input for WrapWithSafeLetCall. Nullable expression '${nullableExpression.text}' should be a descendant" +
+                            " of '${targetExpression.text}'."
+                )
+                return@applyTo
+            }
+            val suggestedVariableName = input.suggestedVariableName
+            val factory = KtPsiFactory(targetExpression)
+
+            fun getNewExpression(nullableExpressionText: String, expressionUnderLetText: String): KtExpression {
+                return when (suggestedVariableName) {
+                    "it" -> factory.createExpressionByPattern("$0?.let { $1 }", nullableExpressionText, expressionUnderLetText)
+                    else -> factory.createExpressionByPattern(
+                        "$0?.let { $1 -> $2 }",
+                        nullableExpressionText,
+                        suggestedVariableName,
+                        expressionUnderLetText
+                    )
+                }
+            }
+
+            val callExpression = nullableExpression.parentOfType<KtCallExpression>(withSelf = true)
+            val qualifiedExpression = callExpression?.getQualifiedExpressionForSelector()
+            val receiverExpression = qualifiedExpression?.receiverExpression
+            if (receiverExpression != null && input.isImplicitInvokeCallToMemberProperty) {
+                // In this case, the nullable expression is an invocable member. For example consider the following
+                //
+                // interface Foo {
+                //   val bar: (() -> Unit)?
+                // }
+                // fun test(foo: Foo) {
+                //   foo.bar()
+                // }
+                //
+                // In this case, `foo.bar` is nullable and this fix should change the code to `foo.bar?.let { it() }`. But note that
+                // the PSI structure of the above code is
+                //
+                // - qualifiedExpression: foo.bar()
+                //   - receiver: foo
+                //   - operationTokenNode: .
+                //   - selectorExpression: bar()
+                //     - calleeExpression: bar
+                //     - valueArgumentList: ()
+                //
+                // So we need to explicitly construct the nullable expression text `foo.bar`.
+                val nullableExpressionText =
+                    "${receiverExpression.text}${qualifiedExpression.operationSign.value}${nullableExpression.text}"
+                val newInvokeCallText = "${suggestedVariableName}${callExpression.valueArgumentList?.text ?: ""}"
```

Comment:
IIUC `callExpression.valueArgumentList` will be null if you have something like:
```kt
interface Foo {
    val bar: (((Int) -> Int) -> Unit)?
}

fun test(foo: Foo) {
    foo.bar {
        val x = it + 2
        x
    }
}
```
I.e., the invocable member accepts a single lambda argument. If this is correct, could you add a test for this?

Reply:
Thanks! Fixed and added a test!